### ISSUE TITLE
u64 might not be enough for higher difficulty

### ISSUE THREAD

**Issue by neeboo** [suidouble/sui_meta#3]
Would you  consider to use U256?

```rust
    public struct Block has key, store {
        nonce: u64,
        ...,
    }

```

**Comment by jeka-kiselyov** [comment]
@neeboo good point, I was thinking about this too. But take in account that there's payload and meta involved in hashing. So if u64 will be not enough - you can just push something there  (actually anything) - and try nonces again in the u64 range. So we are ok here. BTC has the same and it's ok just because of this - payload.
